chore: Run tests in parallel -- Will fail [DO NOT MERGE]

kevaundray opened the pull request:
# Related issue(s)

<!-- If it does not already exist, first create a GitHub issue that describes the problem this Pull Request (PR) solves before creating the PR and link it here. -->

Resolves # <!-- link to issue -->

# Description

Using this as barometer to see how much we look to gain.

## Summary of changes

<!-- Describe the changes in this PR. Point out breaking changes if any. -->

## Dependency additions / changes

<!-- If applicable. -->

## Test additions / changes

<!-- If applicable. -->

# Checklist

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [ ] I have reviewed the changes on GitHub, line by line.
- [ ] I have ensured all changes are covered in the description.

## Documentation needs
- [ ] This PR requires documentation updates when merged.

<!-- If checked, list / describe what needs to be documented. -->

# Additional context

<!-- If applicable. -->

TomAFrench commented:
CI's locked to a [single thread](https://github.com/noir-lang/.github/blob/5fe60d83d7dc83925f1375e9ac5fce11f595bb4d/.github/workflows/rust-test.yml#L79) btw so adding rayon here doesn't do anything.

kevaundray commented:
> CI's locked to a [single thread](https://github.com/noir-lang/.github/blob/5fe60d83d7dc83925f1375e9ac5fce11f595bb4d/.github/workflows/rust-test.yml#L79) btw so adding rayon here doesn't do anything.

Ahh totally forgot

kevaundray commented:
> Ah, you managed to catch that before I deleted it. I'm a little unsure that this is relevant actually. That flag is to tell Rust to run tests sequentially but that doesn't have to mean that each test is itself singlethreaded.

For Rayon its usually `RAYON_NUM_THREADS=1` but I changed it just incase as Rayon may read that environment variable too
